instrumentation/spring/spring-ws-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/ws/SpringWsTracer.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.ws;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.lang.reflect.Method;
+
+public class SpringWsTracer extends BaseTracer {
+
+  private static final SpringWsTracer TRACER = new SpringWsTracer();
+
+  public static SpringWsTracer tracer() {
+    return TRACER;
+  }
+
+  public Span startSpan(Method method) {
+    Span springWsSpan = super.startSpan(method);
+    springWsSpan.setAttribute(
+        SemanticAttributes.CODE_NAMESPACE, method.getDeclaringClass().getName());
+    springWsSpan.setAttribute(SemanticAttributes.CODE_FUNCTION, method.getName());
```

Comment:
this feels like a "route", should we also update the SERVER span name?

Reply:
It seemed a bit weird to have server span with the exact same name as a nested internal span so I wasn't sure whether it would be a good idea. If you believe server span should also be renamed then I'll do it.

Reply:
good point 👍